[CI] MonitoringWithWatcherRestIT.testThatLocalExporterAddsWatches failures

**Build scan**:

https://gradle-enterprise.elastic.co/s/7ozkt4tgixntq

**Repro line**:
```
./gradlew ':x-pack:plugin:watcher:qa:with-monitoring:integTestRunner' --tests "org.elasticsearch.smoketest.MonitoringWithWatcherRestIT.testThatLocalExporterAddsWatches" \
  -Dtests.seed=498D6AE9762D2D55 \
  -Dtests.security.manager=true \
  -Dtests.locale=es-US \
  -Dtests.timezone=Pacific/Efate \
  -Druntime.java=11
```



**Reproduces locally?**:

Wasn't able to reproduce locally within a few runs

**Error**

```
java.lang.AssertionError: 
Expected: is <6>
     but: was <1>
	at __randomizedtesting.SeedInfo.seed([498D6AE9762D2D55:AE7A5AAF2E976FD1]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.smoketest.MonitoringWithWatcherRestIT.lambda$assertTotalWatchCount$1(MonitoringWithWatcherRestIT.java:92)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:929)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:902)
```

**Applicable branches**:

Failure in 7.8, but found similar failures in the past on master, e.g. https://gradle-enterprise.elastic.co/s/wfknexh53ho5m

**Failure history**:

Looks like this failed occasionally (~ 6 times in June) some earlier in May... Latest one on master

https://gradle-enterprise.elastic.co/s/wfknexh53ho5m

A new failure on master: https://gradle-enterprise.elastic.co/s/fseuecrqhx37m

```
java.lang.AssertionError: 
Expected: is <6>
     but: was <1>
	at __randomizedtesting.SeedInfo.seed([5A9BC3BCD5B301F6:BD6CF3FA8D094372]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.smoketest.MonitoringWithWatcherRestIT.lambda$assertTotalWatchCount$2(MonitoringWithWatcherRestIT.java:109)
```

The test is failing once or twice a week on 7.10, 7.x and master

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(build.branch),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testThatLocalExporterAddsWatches),sort:!(process.time-start,desc))

--- a/x-pack/plugin/watcher/qa/with-monitoring/src/javaRestTest/java/org/elasticsearch/smoketest/MonitoringWithWatcherRestIT.java
+++ b/x-pack/plugin/watcher/qa/with-monitoring/src/javaRestTest/java/org/elasticsearch/smoketest/MonitoringWithWatcherRestIT.java
@@ -44,28 +44,6 @@ public class MonitoringWithWatcherRestIT extends ESRestTestCase {
         deleteAllWatcherData();
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/59132")
-    public void testThatLocalExporterAddsWatches() throws Exception {
-        String watchId = createMonitoringWatch();
-
-        Request request = new Request("PUT", "/_cluster/settings");
-        request.setJsonEntity(
-            Strings.toString(
-                jsonBuilder().startObject()
-                    .startObject("persistent")
-                    .field("xpack.monitoring.exporters.my_local_exporter.type", "local")
-                    .field("xpack.monitoring.exporters.my_local_exporter.cluster_alerts.management.enabled", true)
-                    .endObject()
-                    .endObject()
-            )
-        );
-        adminClient().performRequest(request);
-
-        assertTotalWatchCount(WATCH_IDS.length);
-
-        assertMonitoringWatchHasBeenOverWritten(watchId);
-    }
-
     private void assertMonitoringWatchHasBeenOverWritten(String watchId) throws Exception {
         assertBusy(() -> {
             ObjectPath path = ObjectPath.createFromResponse(client().performRequest(new Request("GET", "/_watcher/watch/" + watchId)));

